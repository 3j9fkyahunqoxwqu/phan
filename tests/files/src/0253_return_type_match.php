<?php
class C253_1 {
    /** @return string */
    public function f1() : int {
        return 7;
    }
    /** @return string[] */
    public function f2() : array {
        return array('a', 'b', 'c');
    }
    /** @return $this */
    public function f3() : C253_1 {
        return $this;
    }
    /** @return static */
    public static function f4() : C253_1 {
        return new static();
    }
}
class C253_2 extends C253_1 {
    /** @return C253_2 */
    public function f5() : C253_1 {
        return new C253_2;
    }
    /** @return C253_1 */
    public function f6() : C253_2 {
        return new C253_2;
    }
}
