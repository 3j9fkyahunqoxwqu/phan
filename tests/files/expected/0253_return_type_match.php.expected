%s:4 PhanTypeMismatchDeclaredReturn Doc-block declares return type string which is incompatible with the return type int declared in the signature
%s:26 PhanTypeMismatchDeclaredReturn Doc-block declares return type \C253_1 which is incompatible with the return type \C253_2 declared in the signature
